### PR TITLE
[FLINK-8341] [flip6] Remove not needed options from CommandLineOptions

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

Remove not needed options from `CommandLineOptions`

This PR is based on #5226.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala [hunk 1]
```diff
@@ -253,16 +253,22 @@ object FlinkShell {
     yarnConfig.queue.foreach((queue) => args ++= Seq("-yqu", queue.toString))
     yarnConfig.slots.foreach((slots) => args ++= Seq("-ys", slots.toString))
 
-    val options = CliFrontendParser.parseRunCommand(args.toArray)
+    val commandLine = CliFrontendParser.parse(
+      CliFrontendParser.getRunCommandOptions,
+      args.toArray,
+      true)
+
+    val options = new RunOptions(commandLine)
```
Comment (GJL) [inline flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala]: Unused variable.

File: flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala [hunk 2]
```diff
@@ -253,16 +253,22 @@ object FlinkShell {
...
+
     val frontend = new CliFrontend(
       configuration,
       CliFrontend.loadCustomCommandLines(configuration, configurationDirectory))
     val config = frontend.getConfiguration
```
Comment (GJL) [inline flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala]: Unused variable `config`.

File: flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala [hunk 3]
```diff
@@ -281,16 +287,22 @@ object FlinkShell {
       "-m", "yarn-cluster"
     )
 
-    val options = CliFrontendParser.parseRunCommand(args.toArray)
+    val commandLine = CliFrontendParser.parse(
+      CliFrontendParser.getRunCommandOptions,
+      args.toArray,
+      true)
+
+    val options = new RunOptions(commandLine)
```
Comment (GJL) [inline flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala]: Unused variable  `options`.

File: flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala [hunk 4]
```diff
@@ -281,16 +287,22 @@ object FlinkShell {
...
+
     val frontend = new CliFrontend(
       configuration,
       CliFrontend.loadCustomCommandLines(configuration, configurationDirectory))
     val config = frontend.getConfiguration
```
Comment (GJL) [inline flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala]: Unused variable `config`.

**Comment by tillrohrmann** [comment]
Thanks for the review @GJL. Addressing comments and merging once Travis gives green light.

**Pull request closed without merging**